pkg/auth/azure_auth.go

```diff
@@ -100,9 +100,17 @@ func GetServicePrincipalToken(config *AzureAuthConfig, env *azure.Environment, r
 		}
 		if len(config.UserAssignedIdentityID) > 0 {
 			klog.V(4).Info("azure: using User Assigned MSI ID to retrieve access token")
-			return adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(msiEndpoint,
-				resource,
-				config.UserAssignedIdentityID)
+			if config.UserAssignedIdentityID[0] == '/' {
```

Comment:
what about changing the validation to something like below:

```go
var isResourceID = regexp.MustCompile(
	`(?i)/subscriptions/(.+?)/resourcegroups/(.+?)/providers/Microsoft..*`).MatchString

if if isResourceID(config.UserAssignedIdentityID) {...}
```

Reply:
Good idea. Updated to use a standard function `azure.ParseResourceID()`